treemap/script.js

```diff
@@ -87,211 +74,192 @@ d3.json('data.json', function(err, dataset) {
   // Player names
   cell
     .append('text')
-    .attr('clip-path', function(d) {
-      return 'url(#clip-' + d.data.id + ')';
-    })
+    .attr('clip-path', d => 'url(#clip-' + d.data.id + ')')
     .append('tspan')
-    .attr('x', 8)
-    .attr('y', 8)
-    .attr('dy', '.8em')
-    .attr('class', 'playerNames')
+    .at({
+      x: 8,
+      y: 8,
+      dy: '.8em',
+      class: 'playerNames',
+    })
     .text(function(d) {
       // Only display text if sibling <rect> element is wide enough
-      var parentRect = this.parentNode.previousElementSibling;
+      const parentRect = this.parentNode.previousElementSibling;
       if (d3.select(parentRect).classed('wide')) {
         return d.data.name;
       }
     });
 
   cell
     .append('text')
-    .attr('clip-path', function(d) {
-      return 'url(#clip-' + d.data.id + ')';
-    })
+    .attr('clip-path', d => 'url(#clip-' + d.data.id + ')')
     .append('tspan')
-    .attr('x', 8)
-    .attr('y', 20)
-    .attr('dy', '1.2em')
-    .attr('class', 'playerNamesFee')
+    .at({
+      x: 8,
+      y: 20,
+      dy: '1.2em',
+      class: 'playerNamesFee',
+    })
     .text(function(d) {
       // Only display text if sibling <rect> element is wide enough
-      var parentRect = this.parentNode.parentNode.firstElementChild;
+      const parentRect = this.parentNode.parentNode.firstElementChild;
       if (d3.select(parentRect).classed('wide')) {
         return '£' + d.data.fee.split('.')[0] + 'm';
       }
     });
 
   // 💩 Manual labelling
-  var key = [
+  const key = [
     { name: 'Europe', color: '#254251' },
     { name: 'Britain', color: '#E0AB26' },
     { name: 'Africa', color: '#3292A6' },
     { name: 'S. America', color: '#F37F2F' },
   ];
 
-  var legendx = config.width < 400 ? 10 : width + 10;
-  var legendy = config.width < 400 ? 310 : 10;
-  var legendheight = config.width < 400 ? 10 : 20;
+  let legendConfig = {
+    x: width + 10,
+    y: 10,
+    height: 20,
+  };
 
   // Create a legend element
-  var legend = container
+  const legend = container
     .append('g')
-    .attr('class', 'legendContainer')
+    .at({ class: 'legendContainer' })
     .selectAll('g')
     .data(key)
     .enter()
     .append('g')
-    .attr('class', 'legend')
-    .attr('transform', function(d, i) {
-      var leftmargin = 0;
-      var topmargin = 0;
-      var x = leftmargin + legendx;
-      var y = i * legendheight + legendy + topmargin;
+    .at({ class: 'legend' })
+    .attr('transform', (d, i) => {
+      const leftmargin = 0;
+      const topmargin = 0;
+      const x = leftmargin + legendConfig.x;
+      const y = i * legendConfig.height + legendConfig.y + topmargin;
       return 'translate(' + x + ',' + y + ')';
```

Comment:
Is this easier? Four `+` and multiple `'` is confusing

```
return `translate(${x},${y})`;
```

Reply:
fixed but doesn't show